src/compas/datastructures/_mixins/filters.py

```diff
@@ -78,6 +90,62 @@ def vertices_where(self, conditions, data=False):
                 else:
                     yield key
 
+    def vertices_where_predicate(self, predicate, data=False):
+        """Get vertices for which a certain condition or set of conditions is true using a lambda function.
+
+        Parameters
+        ----------
+        predicate : function
+            The condition you want to evaluate.
+        data : bool, optional
+            Yield the vertices and their data attributes.
+            Default is ``False``.
+
+        Yields
+        ------
+        key: hashable
+            The next vertex that matches the condition.
+        2-tuple
+            The next vertex and its attributes, if ``data=True``.
+
+        Examples
+        --------
+        .. code-block:: python
+
+            from compas.datastructures import Mesh
+
+            mesh = Mesh()
+
+            v0 = mesh.add_vertex( x = 0.0, y = 0.0, z = 0.0, extra_attr1 = 0, extra_attr2 = [])
+            v1 = mesh.add_vertex( x = 100.0, y = 0.0, z = 0.0, extra_attr1 = 2, extra_attr2 = [3,5,9])
+            v2 = mesh.add_vertex( x = 100.0, y = 100.0, z = 0.0, extra_attr1 = 1, extra_attr2 = ['value1', 'value2'])
+            v3 = mesh.add_vertex( x = 0.0, y = 100.0, z = 0.0, extra_attr1 = 2, extra_attr2 = [3,7,12])
+
```

Comment:
please remove white space between the parenthesis.
only white space after the comma there...

Reply:
Done.